### ISSUE TITLE
[Feature Request] Cloudflare Turnstile

### ISSUE THREAD

**Issue by evrifaessa** [FOSSBilling/FOSSBilling#308]
Cloudflare has recently announced their alternative to Google's reCAPTCHA.

We might want to have a look on that. https://www.cloudflare.com/en-gb/products/turnstile/

**Comment by evrifaessa** [comment]
https://blog.cloudflare.com/turnstile-private-captcha-alternative/

**Comment by weismannweb** [comment]
nice, yeah i knew this about google, glad to see theirs an alternative now, i think i will use this for my clients too

**Comment by John-S4** [comment]
When we update our captcha implementation, then it would be much better to use something like hCaptcha than to jump from a Google provided solution to a Cloudflare provided one. 

Closing this as #1658 is a better solution.

**Comment by evrifaessa** [comment]
Neither is a "better option" per se. Let's just focus on removing the hardcoded Google reCaptcha and making way for the other providers.

And please keep this open as a feature request, it doesn't have to be a "milestone" for v1.0, it's just nice to have options.

**Comment by John-S4** [comment]
> Neither is a "better option" per se. Let's just focus on removing the hardcoded Google reCaptcha and making way for the other providers.

I would disagree and say that in this use case, one is a better option than the other, but that's not the big point here I guess. 

Yes, ultimately a modular solution where there were a number of possible captcha options to choose from would be ideal, but that is a much bigger thing than replacing the existing reCaptcha installation with a more up to date implementation of someting, i.e. hCaptcha. 

That should be a relatively straightforward step 1, with implementation of a more modular system as a longer term goal.

**Comment by BelleNottelling** [comment]
> That should be a relatively straightforward step 1, with implementation of a more modular system as a longer term goal.

I get the reasoning behind this, but practically speaking making it modular is most likely a very straightforward process that shouldn't take any more than roughly half an hour to do & it would mean we don't need to eliminate the current reCAPTCHA implementation and instead we'd just need to add any new options along side of it

**Comment by John-S4** [comment]
> [..] practically speaking making it modular is most likely a very straightforward process that shouldn't take any more than roughly half an hour to do & it would mean we don't need to eliminate the current reCAPTCHA implementation and instead we'd just need to add any new options along side of it

In that case I suggest that we make a new issue for modularization and roadmap it for 0.7. - #1662

**Comment by BelleNottelling** [comment]
I don't think we need an open issue for each specific potential captcha implementation.
Closing in favor of #1662
